### PR TITLE
53 app viewmodel

### LINKED ISSUE #53: Implement AppViewModel
This view model will hold the user object as a state. It will pass the user down to ui states for each page.

It should be able to read, update, create and delete from the database using `UserRepository` and `DateRepository`. This branch should complete the integration between Calendar and database.

- [ ] It should determine if the user is a return user or a new user and redirect the user to the correct page
- [ ] Implement functions for CRUD
- [ ] Integrate the UI with the view model

### PR DESCRIPTION
**Pull request by PierreLessard**
This pull request is for the implementation of data persistence, via the appViewModel. 
- The appViewModel grabs user and date data from the repo. 
- The appViewModel now supplies the calendarViewModel with day objects that were previously logged. 
- The log screen now adds new dates to the date Repo via the appViewModel
- Added get and setters for tracked symptoms, so setting screen can access and change tracked symptoms

### CODE REVIEW

**Review by kevinle623 — APPROVED** [review]
Looks great! Clean code. Played around with the calendar options and data storage and works smoothly.

File: app/src/main/java/com/tpp/theperiodpurse/NavigationGraph.kt [hunk 1]
```diff
@@ -130,20 +133,39 @@ fun NavigationGraph(
...
+                    onboardViewModel.addNewDate(
+                        currentDate,
+                        FlowSeverity.Heavy,
+                        Mood.ANGRY,
+                        duration,
+                        Exercise.YOGA,
+                        CrampSeverity.Bad,
+                        duration
+                    )
+                    onboardViewModel.addNewUser(symptomList, dateList, 0, 0, 0)
```
Comment (leowrites) [inline app/src/main/java/com/tpp/theperiodpurse/NavigationGraph.kt]: Can we remove the hard coded lines now?

File: app/src/main/java/com/tpp/theperiodpurse/ui/calendar/CalendarScreen.kt [hunk 2]
```diff
@@ -58,6 +65,114 @@ val tabModifier = Modifier
...
+                }
+                Icon(
+                    painter = painterResource(id = symptom.resourceId),
+                    tint = color,
+                    contentDescription = stringResource(symptom.nameId)
+                )
+            }
+        }
+    }
+}
```
Comment (leowrites) [inline app/src/main/java/com/tpp/theperiodpurse/ui/calendar/CalendarScreen.kt]: Are these duplicated from `calendar/components/SymptomTab.kt` ?

Comment (Manpreet-Bhatti) [reply app/src/main/java/com/tpp/theperiodpurse/ui/calendar/CalendarScreen.kt]: They're not direct copy & paste but they're similar. I believe SymptomTab has more to its functions than the functions used in CalendarScreen. @michaelng0107 is it safe to remove these duplicate functions from CalendarScreen?

**Pull request merged**